### ISSUE TITLE
[dune-configurator] PKG_CONFIG_PATH is not resolved on macos 

### ISSUE THREAD

**Issue by dmtrKovalenko** [ocaml/dune#10805]
## Expected Behavior
When providing custom PKG_CONFIG_PATH variable dependencies

## Actual Behavior
PKG_CONFIG_PATH is ignored in favor of brew https://github.com/ocaml/dune/blob/main/otherlibs/configurator/src/v1.ml

## Specifications

- Version of `dune` (output of `dune --version`): 3.16.0
- Version of `ocaml` (output of `ocamlc --version`): 5.2.0
- Operating system (distribution and version): MacOS 14.5

## Additional information 
 
Despite the code looks correct it is for some reason still not running, would be also good to log the modified PKG_CONFIG_PATH

**Comment by rgrinberg** [comment]
Do you want to send a PR? I don't think many of the dune devs use homebrew, so we need help from somebody that knows about brew and is willing to test things.

**Comment by anmonteiro** [comment]
I don't think this is true since Nix relies pretty heavily on PKG_CONFIG_PATH

**Comment by dmtrKovalenko** [comment]
Yeah I do handle it differently in my projects, so can make a PR porting my changes
